plugins/logistics.cpp

```diff
@@ -382,6 +383,44 @@ class TrainStockProcessor : public StockProcessor {
     }
 };
 
+class ForbidStockProcessor : public StockProcessor {
+public:
+    ForbidStockProcessor(int32_t stockpile_number, bool enabled, ProcessorStats& stats)
+        : StockProcessor("forbid", stockpile_number, enabled, stats) { }
+
+    bool is_designated(color_ostream& out, df::item* item) override {
+        return item->flags.bits.forbid;
+    }
+
+    bool can_designate(color_ostream& out, df::item* item) override {
+        return true;
```

Comment:
you cannot forbid artifacts. this should return `!is_artifact(item)`

Reply:
(which the DumpStockProcessor should also be checking for -- could you add that check there too?)

Reply:
Oh! I didn’t know that, thanks!